File: ios/NotificationHelper.swift
import Foundation
import UserNotifications

/// Posts a local notification when a non-empty diagnostic batch drains. If the
/// app lacks notification authorization the request silently no-ops — never a
/// crash, never a forced prompt.
enum NotificationHelper {
  static func post(count: Int) {
    let center = UNUserNotificationCenter.current()
    let content = UNMutableNotificationContent()
    content.title = "New diagnostics captured"
    content.body = count == 1 ? "1 diagnostic is available." : "\(count) diagnostics are available."
    let request = UNNotificationRequest(
      identifier: UUID().uuidString, content: content, trigger: nil)
    center.add(request, withCompletionHandler: nil)
  }
}

File: android/src/main/java/com/metrickit/NotificationHelper.kt
package com.metrickit

import android.app.NotificationChannel
import android.app.NotificationManager
import android.content.Context
import android.os.Build

/**
 * Posts a local notification on a non-empty drain. Best-effort: if the host app
 * lacks the runtime notification permission (Android 13+) the post silently
 * no-ops. Never throws.
 */
internal object NotificationHelper {
  private const val CHANNEL_ID = "react_native_metrickit_diagnostics"
  private const val CHANNEL_NAME = "Diagnostics"
  private const val NOTIFICATION_ID = 9317

  fun post(context: Context, count: Int) {
    SafeExecutor.attempt("NotificationHelper.post") {
      val manager =
        context.getSystemService(Context.NOTIFICATION_SERVICE) as? NotificationManager ?: return@attempt

      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
        val channel = NotificationChannel(
          CHANNEL_ID, CHANNEL_NAME, NotificationManager.IMPORTANCE_LOW
        )
        manager.createNotificationChannel(channel)
      }

      val builder = android.app.Notification.Builder(context, CHANNEL_ID)
        .setContentTitle("New diagnostics captured")
        .setContentText(
          if (count == 1) "1 diagnostic is available." else "$count diagnostics are available."
        )
        .setSmallIcon(context.applicationInfo.icon)

      manager.notify(NOTIFICATION_ID, builder.build())
    }
  }
}
